examples/asset/alter_mesh.rs

```diff
@@ -0,0 +1,228 @@
+//! Shows how to modify mesh assets after spawning.
+
+use bevy::{gltf::GltfLoaderSettings, input::common_conditions::input_just_pressed, prelude::*};
+use bevy_render::{mesh::VertexAttributeValues, render_asset::RenderAssetUsages};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, (setup, spawn_text))
+        .add_systems(
+            Update,
+            alter_handle.run_if(input_just_pressed(KeyCode::Space)),
+        )
+        .add_systems(
+            Update,
+            alter_mesh.run_if(input_just_pressed(KeyCode::Enter)),
+        )
+        .run();
+}
+
+#[derive(Component, Debug)]
+enum Shape {
+    Cube,
+    Sphere,
+}
+
+impl Shape {
+    fn get_model_path(&self) -> String {
+        match self {
+            Shape::Cube => "models/cube/cube.gltf".into(),
+            Shape::Sphere => "models/sphere/sphere.gltf".into(),
+        }
+    }
+
+    fn set_next_variant(&mut self) {
+        *self = match self {
+            Shape::Cube => Shape::Sphere,
+            Shape::Sphere => Shape::Cube,
+        }
+    }
+}
+
+#[derive(Component, Debug)]
+struct Left;
+
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let left_shape = Shape::Cube;
+    let right_shape = Shape::Cube;
+
+    // In normal use, you can call `asset_server.load`, however see below for an explanation of
+    // `RenderAssetUsages`.
+    let left_shape_model = asset_server.load_with_settings(
+        GltfAssetLabel::Primitive {
+            mesh: 0,
+            // This field stores an index to this primitive in its parent mesh. In this case, we
+            // want the first one. You might also have seen the syntax:
+            //
+            //     models/cube/cube.gltf#Scene0
+            //
+            // which accomplishes the same thing.
+            primitive: 0,
+        }
+        .from_asset(left_shape.get_model_path()),
+        // This is the default loader setting, ensuring both `RENDER_WORLD` and `MAIN_WORLD` are
+        // enabled. It's provided explicitly here only by way of demonstration.
+        //
+        // A common mistake is to use `RENDER_WORLD` by itself, which can cause a confusing lack of
+        // assets available in `Res<Assets<Mesh>>`. `RENDER_WORLD` alone will cause the asset to be
+        // unloaded from the asset server after it's been sent to the GPU. Unless you have a clear
+        // reason not to do so, it's best to use `RenderAssetUsages::all().
+        |settings: &mut GltfLoaderSettings| settings.load_meshes = RenderAssetUsages::all(),
```

Comment:
@JMS55 Is this the kind of thing you had in mind? New territory for me, so feel free to provide more direction!

Reply:
@JMS55 will know better than me, but IME it is the other way around: usually you won't be mutating meshes at runtime, so it's best to leave it at RENDER_WORLD only unless you have a good reason to set it to ALL.

Reply:
So JMS55 wrote:
> but I'd prefer if we could explicitly set that to all() and call out why it's necessary, as I very frequently see people make the mistake of using RENDER_WORLD only and then being confused why their asset does not exist.
Perhaps if I make a point of only setting this to `all` for the asset whose mesh we're going to directly modify, and re-word the comment to say the default is "usually safe" unless you have more specific needs? I'll push that so you can have a read.

Reply:
The default is *always* safe, as the default is ALL. Many people choose to use RENDER_WORLD instead of ALL for meshes, which is safe until you need to access the mesh on the CPU, which we do in this example.

Reply:
I'm with you now. OK, so maybe the second one we explicitly set it to `RENDER_WORLD` and add a further comment there. Maybe also modify the sprite example.